### PR TITLE
refactor(core): auth

### PR DESCRIPTION
**Pull request by forehalo**
closes AF-753 AF-1227

### CODE REVIEW

**Comment by forehalo** [comment]
* **#7999** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7999?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#8019](https://github.com/toeverything/AFFiNE/pull/8019) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8019?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#8032](https://github.com/toeverything/AFFiNE/pull/8032) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8032?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈
* **#7994** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7994?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @forehalo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by EYHN — APPROVED** [review]

File: packages/frontend/electron/src/main/deep-link.ts [hunk 1]
```diff
@@ -61,51 +61,28 @@ async function handleAffineUrl(url: string) {
...
     const mainWindow = await getMainWindow();
     if (mainWindow) {
       mainWindow.show();
     }
+  } else {
+    await openUrl(urlObj);
   }
 }
 
-async function handleOauthJwt(url: string) {
```
Comment (pengx17) [inline packages/frontend/electron/src/main/deep-link.ts:77]: what if user uses old client but login with the new server's auth impl?

Comment (forehalo) [reply packages/frontend/electron/src/main/deep-link.ts:77]: the '/desktop-signin' route is kept in frontend and old client's request send it as `redirect_uri`. there should be any problem. I will test it anyway

**Review by pengx17 — APPROVED** [review]
looks like we are lacking of user prompts when loggin successfully or failed.

**Comment by forehalo** [comment]
> looks like we are lacking of user prompts when loggin successfully or failed.

there should be a place monitoring the 'authenticated' status and toast

**Review by Brooooooklyn — APPROVED** [review]

**Review by darkskygit — APPROVED** [review]

**Pull request merged**